protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/deployment/DecisionRequirementsRecord.java

```diff
@@ -0,0 +1,134 @@
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.protocol.impl.record.value.deployment;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.bufferAsArray;
+import static io.camunda.zeebe.util.buffer.BufferUtil.bufferAsString;
+
+import io.camunda.zeebe.msgpack.property.BinaryProperty;
+import io.camunda.zeebe.msgpack.property.IntegerProperty;
+import io.camunda.zeebe.msgpack.property.LongProperty;
+import io.camunda.zeebe.msgpack.property.StringProperty;
+import io.camunda.zeebe.protocol.impl.record.UnifiedRecordValue;
+import io.camunda.zeebe.protocol.record.value.deployment.DecisionRequirementsRecordValue;
+import org.agrona.DirectBuffer;
+
+public final class DecisionRequirementsRecord extends UnifiedRecordValue
+    implements DecisionRequirementsRecordValue {
+
+  private final StringProperty decisionRequirementsIdProp =
+      new StringProperty("decisionRequirementsId");
+  private final StringProperty decisionRequirementsNameProp =
+      new StringProperty("decisionRequirementsName");
+  private final IntegerProperty decisionRequirementsVersionProp =
+      new IntegerProperty("decisionRequirementsVersion");
+  private final LongProperty decisionRequirementsKeyProp =
+      new LongProperty("decisionRequirementsKey");
+  private final StringProperty namespaceProp = new StringProperty("namespace");
+
+  private final StringProperty resourceNameProp = new StringProperty("resourceName");
+  private final BinaryProperty checksumProp = new BinaryProperty("checksum");
+  private final BinaryProperty resourceProp = new BinaryProperty("resource");
+
+  public DecisionRequirementsRecord() {
+    declareProperty(decisionRequirementsIdProp)
+        .declareProperty(decisionRequirementsNameProp)
+        .declareProperty(decisionRequirementsVersionProp)
+        .declareProperty(decisionRequirementsKeyProp)
+        .declareProperty(namespaceProp)
+        .declareProperty(resourceNameProp)
+        .declareProperty(checksumProp)
+        .declareProperty(resourceProp);
+  }
+
+  @Override
+  public String getDecisionRequirementsId() {
+    return bufferAsString(decisionRequirementsIdProp.getValue());
+  }
+
+  @Override
+  public String getDecisionRequirementsName() {
+    return bufferAsString(decisionRequirementsNameProp.getValue());
+  }
+
+  @Override
+  public int getDecisionRequirementsVersion() {
+    return decisionRequirementsVersionProp.getValue();
+  }
+
+  @Override
+  public long getDecisionRequirementsKey() {
+    return decisionRequirementsKeyProp.getValue();
+  }
+
+  @Override
+  public String getNamespace() {
+    return bufferAsString(namespaceProp.getValue());
+  }
+
+  @Override
+  public String getResourceName() {
+    return bufferAsString(resourceNameProp.getValue());
+  }
+
+  @Override
+  public byte[] getChecksum() {
+    return bufferAsArray(checksumProp.getValue());
+  }
+
+  @Override
+  public boolean isDuplicate() {
+    return false;
+  }
```

Comment:
❓ Do we want to always consider these records duplicate? Or did you implement this as a first iteration, and we can implement duplication checks for it later?

Reply:
We will implement the duplication checks later with #8174.

However, the records will always be no duplicates. We write the records only for new versions. 

But we will use the property for the new deployment command #8065. If a DMN is already deployed then it is added to the deployment response and marked as duplicate. Same as for processes.

I added the property already to avoid later changes in the records and ES templates.